### PR TITLE
feat: enable opbeans java release process

### PR DESCRIPTION
**Pull request by v1v**
## Highlights
- enable ITs: async when running as a PR or sync when running in a branch/tag
- Continuous delivery when running the pipeline either in a branch or tag. master branch refers to `latest` while others will name the image with their names.
- Script to bump versions for the java agent.

## Issues
- Depends on:
  - https://github.com/elastic/apm-integration-testing/pull/556
  - https://github.com/elastic/apm-pipeline-library/pull/271
  - https://github.com/elastic/apm-agent-java/pull/940

## Tasks
- [x] Versioning
- [x] Release automation
- [x] ITs validation should be synchronous when running a release.

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #!/usr/bin/env groovy
-@Library('apm@current') _
+@Library('apm@opbeans-release') _
```
Comment (v1v) [inline .ci/Jenkinsfile]: This should be reverted once the tests have been done

**Comment by felixbarny** [comment]
Sorry if it's obvious but what is the benefit of having a release of Opbeans java with each release of the Java agent? Which problems are we trying to solve here? Is it just to test that the agent API is binary compatible?

**Comment by kuisathaverat** [comment]
>Sorry if it's obvious but what is the benefit of having a release of Opbeans java with each release of the Java agent? Which problems are we trying to solve here? Is it just to test that the agent API is binary compatible?

have a Docker image with the agent and the application in the same binary allow to make offline test, you only need the Docker image nothing else. it avoids having runtime dependencies to download, and all the problems derivated that, you have only one download. Have a well-defined version named allow to have a compatibility matrix of the Opbeans with the stack, and make super easy to test any version of the agent against any version of the start, you only specify the version of the agent, and in the future a version attached to the stack version, something like `opbeans/opbeans-java:1.0` and `opbeans/opbeans-java:stack-7.5`. To have well-defined versioning allows us to track issues related to an opbeans version and a version of the stack. Finally, it allows to have decupled versions of the frontend and the backend, the RUM agent is gonna start a series of changes that breaks compatibility, this means that all Opbeans can stop working if they use the `latest` version of the frontend, with well-defined versioning this does not happen.

File: .ci/Jenkinsfile (hunk 1 above)
Comment (v1v) [inline .ci/Jenkinsfile]:
```suggestion
@Library('apm@current') _
```

**Pull request merged**